compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/inline/FunctionInlining.kt

```diff
@@ -227,6 +229,13 @@ class FunctionInlining(
             }
         }
 
+        private fun saveParameterToExpressionMapping(callee: IrFunction, copiedCallee: IrFunction) {
+            for ((parameter, copiedParameter) in callee.valueParameters.zip(copiedCallee.valueParameters)) {
+                val expression = substituteMap[copiedParameter] ?: continue
+                context.mapping.inlinedParameterToExpression[parameter] = expression
```

Comment:
Did I understand correctly that you are using the inline function parameter as the key in the map? How it's supposed to work if the function is called several times? Can you please add a test on this case, if I'm missing something and it works?